eng/Versions.props

```diff
@@ -43,13 +43,13 @@
   <PropertyGroup>
     <!-- For source generator support we need to target multiple versions of Rolsyn in order to be able to run on older versions of Roslyn -->
     <MicrosoftCodeAnalysisCSharpWorkspacesVersion_3_11>3.11.0</MicrosoftCodeAnalysisCSharpWorkspacesVersion_3_11>
-    <MicrosoftCodeAnalysisCSharpWorkspacesVersion>4.0.0-3.final</MicrosoftCodeAnalysisCSharpWorkspacesVersion>
-    <MicrosoftCodeAnalysisVersion>4.0.0-3.final</MicrosoftCodeAnalysisVersion>
+    <MicrosoftCodeAnalysisCSharpWorkspacesVersion>4.0.0-4.final</MicrosoftCodeAnalysisCSharpWorkspacesVersion>
+    <MicrosoftCodeAnalysisVersion>4.0.0-4.final</MicrosoftCodeAnalysisVersion>
   </PropertyGroup>
   <PropertyGroup>
     <!-- Code analysis dependencies -->
-    <MicrosoftCodeAnalysisCSharpCodeStyleVersion>3.10.0</MicrosoftCodeAnalysisCSharpCodeStyleVersion>
-    <MicrosoftCodeAnalysisCSharpVersion>3.10.0</MicrosoftCodeAnalysisCSharpVersion>
+    <MicrosoftCodeAnalysisCSharpCodeStyleVersion>4.0.0-4.final</MicrosoftCodeAnalysisCSharpCodeStyleVersion>
```

Comment:
@sharwell which version of VS is required when using the updated version? I assume 2022?

Reply:
I assume so.

Reply:
@ViktorHofer, do you have concerns?

Reply:
> which version of VS is required when using the updated version?

2022 Preview 4

Reply:
The docs list VS 2019 as a supported version: https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/windows-requirements.md#visual-studio. Would make sense to update them.

Reply:
A few lines earlier we're changing several 4.0.0-3.final versions to 4.0.0-4.final.  Isn't 4.0.0-3.final also VS2022, just Preview 3?

Regardless, yes, I can update the Windows requirements.  Folks should be using VS2022.  We should make sure that we never require a preview that hasn't been released yet, but I think beyond that is fair game.

Reply:
We probably already require VS 2022 but just forgot to update the docs. Agreed that we shouldn't depend on a VS version not yet released but preview is fine.